amplify/function/BusinessLogic/AddUserToGroup/handler.ts

```diff
@@ -0,0 +1,132 @@
+import { Amplify } from "aws-amplify";
+import { generateClient } from "aws-amplify/data";
+
+import {
+  AdminAddUserToGroupCommand,
+  AdminListGroupsForUserCommand,
+  AdminRemoveUserFromGroupCommand,
+  CognitoIdentityProviderClient,
+} from "@aws-sdk/client-cognito-identity-provider";
+
+import type { Schema } from "../../../data/resource";
+import { updateUser } from "./graphql/mutations";
+
+Amplify.configure(
+  {
+    API: {
+      GraphQL: {
+        endpoint: process.env.AMPLIFY_DATA_GRAPHQL_ENDPOINT as string,
+        region: process.env.AWS_REGION,
+        defaultAuthMode: "iam",
+      },
+    },
+  },
+  {
+    Auth: {
+      credentialsProvider: {
+        getCredentialsAndIdentityId: async () => ({
+          credentials: {
+            accessKeyId: process.env.AWS_ACCESS_KEY_ID as string,
+            secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY as string,
+            sessionToken: process.env.AWS_SESSION_TOKEN as string,
+          },
+        }),
+        clearCredentialsAndIdentityId: () => {
+          /* noop */
+        },
+      },
+    },
+  },
+);
+
+type Handler = Schema["addUserToGroup"]["functionHandler"];
+
+const client = new CognitoIdentityProviderClient({});
+
+const dynamoClient = generateClient<Schema>();
+
+export const handler: Handler = async (event) => {
+  try {
+    const { userId, groupName } = event.arguments;
+
+    // Get the user current group
+    const listGroupCommand = new AdminListGroupsForUserCommand({
+      Username: userId,
+      UserPoolId: process.env.AMPLIFY_AUTH_USERPOOL_ID as string,
+    });
+    const listGroupResponse = await client.send(listGroupCommand);
+
+    if (listGroupResponse.Groups?.length === 0) {
```

Comment:
Is this situation really an error? I think it means that the user just isn't part of a group in which case we don't want to remove them from a group, but I don't think we need to return a 404 error and can proceed with adding them to the requested group